nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -1662,6 +1673,11 @@ private Description handleInvocation(
       // This statement should be unreachable without assigning actual beforehand:
       Preconditions.checkNotNull(actual);
       // make sure we are passing a non-null value
+      if (config.isJSpecifyMode()) {
+        mayActualBeNull =
+            new GenericsChecks(state, config, this)
+                .hasMismatchedNullabilityOfArguments(methodSymbol, actualParams);
+      }
```

Comment:
This is the wrong place to be modifying the logic.  You should be updating `argumentPositionNullness[i]` for each parameter around line 1633, and then the rest of the logic will just work.

Reply:
will update the code. Thanks.